src/pybind/mgr/telemetry/module.py

```diff
@@ -0,0 +1,327 @@
+"""
+Telemetry module for ceph-mgr
+
+Collect statistics from Ceph cluster and send this back to the Ceph project
+when user has opted-in
+"""
+import errno
+import json
+import re
+import requests
+import uuid
+import time
+from datetime import datetime
+from threading import Event
+from collections import defaultdict
+
+from mgr_module import MgrModule
+
+
+class Module(MgrModule):
+    run = False
+    last_upload = None
+    last_report = dict()
+    config = dict()
+    report_id = None
+
+    metadata_keys = [
+            "arch",
+            "ceph_version",
+            "os",
+            "cpu",
+            "kernel_description",
+            "kernel_version",
+            "distro_description",
+            "distro"
+    ]
+
+    config_keys = {
+        'url': 'https://telemetry.ceph.com/report',
+        'enabled': True,
+        'leaderboard': False,
+        'description': None,
+        'contact': None,
+        'organization': None,
+        'proxy': None,
+        'interval': 72
+    }
+
+    COMMANDS = [
```

Comment:
Need to add an OPTIONS definition here (see message to list "mgr module interface change in mimic: config vs. store separation"), since the new mgr config stuff landed.